feature(get_scylla_version): output scylla full version

amoskong opened the pull request:
Currently we only output the major version (eg: 4.1.2), the full version
(contains the build data and commit prefix) of each node won't be
outputed.

This patch changed to output detailed version by info level, then we can see it
early in jenkins log.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

fruch commented:
@amoskong 
check the unittests, it broken cause of this change:
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-2665/2/execution/node/32/log/

fruch commented:
this change might be problematic for some of the code, since calls to this function might be expecting specific format of the version.

also make sure `SCYLLA_VERSION_RE` is matching this version, since without that you'll break and no version would be saved.

fruch reviewed (changes requested):
If this only for having this printed in the logs or email, I'll be more careful, or add a new function.
since this one can affect logic across SCT code.

amoskong commented:
> this change might be problematic for some of the code, since calls to this function might be expecting specific format of the version.
> 
> also make sure `SCYLLA_VERSION_RE` is matching this version, since without that you'll break and no version would be saved.

I manually tested by SCYLLA_VERSION_RE in preparing the test, the changed rpm query format worked well as the output of `scylla --version`.

So this change won't effect the return value of get_scylla_version().

```
[amos@amos-centos7 sct-master]$ scylla --version
4.1.2-0.20200715.cd4502ee64
[amos@amos-centos7 sct-master]$ rpm --query --queryformat '%{VERSION}' scylla
4.1.2[amos@amos-centos7 sct-master]$ 
[amos@amos-centos7 sct-master]$ rpm --query --queryformat '%{VERSION}-%{RELEASE}' scylla
4.1.2-0.20200715.cd4502ee64[amos@amos-centos7 sct-master]$ 


4.1.2 will be matched by SCYLLA_VERSION_RE and returned.
```

amoskong commented:
> If this only for having this printed in the logs or email, I'll be more careful, or add a new function.
> since this one can affect logic across SCT code.

This patch didn't added some thing new, but add something thing missing in the past.
After this change, the outputs of `scylla --version`, `rpm --query ...` and `dpkg-query --show ...` will be more similar than past.

I'm fine to add a new function and don't reuse the code in get_scylla_version().

amoskong commented:
I saw we just added support to include the detailed version to report email: 
- https://github.com/scylladb/scylla-cluster-tests/commit/eb49163d4f7af5a37adb9349c5edd2579657a7c4

Just changed this PR only output the detailed version by info level.

fruch reviewed (approved):
LGTM
